tests/Controllers/login.controller.test.ts

```diff
@@ -0,0 +1,68 @@
+import { loginController } from '../../src/Controllers/login.controller';
+import { TypedRequestBody } from '../../src/Types/TypedRequestBody';
+import { connectToDB } from '../../src/Configs/db.config';
+import { UsersModel } from '../../src/Models/users.model';
+import { User } from '../../src/Types/User';
+import bcrypt from 'bcrypt';
+import mongoose from 'mongoose';
+import { Response } from 'express';
+
+describe('Test of login controller', () => {
+    const mockResponse = () => {
+        const res: Partial<Response> = {};
+        res.json = jest.fn().mockReturnValue(res);
+        res.status = jest.fn().mockReturnValue(res);
+        res.send = jest.fn().mockReturnValue(res);
+        return res;
+    };
+
+    let hashedPassword: string;
+
+    beforeAll(async () => {
+        await connectToDB();
+
+        hashedPassword = await bcrypt.hash('password', 10);
```

Comment:
couldn't password be hashed only once and held on 19 line?

Reply:
unfortunately no, because this` bcrypt.hash` function is asynchronous so in order to await its result it has to be placed inside `beforeAll` async function